File: 백준/Silver/9375. 패션왕 신해빈/패션왕 신해빈.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');
const T = Number(input.shift());

for (let i = 0; i < T; i++) {
    const n = Number(input.shift());  // 각 테스트케이스의 n
    const counter = new Map();
    
    for (let j = 0; j < n; j++) {
        const [name, category] = input.shift().split(' ');
        counter.set(category, (counter.get(category) || 0) + 1);
    }
    
    let result = 1;
    for (const count of counter.values()) {
        result *= (count + 1);
    }
    console.log(result - 1);
}

File: 백준/Silver/9375. 패션왕 신해빈/패션왕 신해빈.py
from collections import Counter

T = int(input())
for _ in range(T):
    n = int(input())
    
    counter = Counter()
    for _ in range(n):
        name, category = input().split()
        counter[category] += 1
        
    result = 1
    for count in counter.values():
        result *= (count + 1)
     
    print(result - 1)